packages/integrations/preact/src/index.ts

```diff
@@ -1,94 +1,71 @@
+// TODO: Add back compat support -- how would this work in the new world?
+export default function ({include, exclude, compat}: Options = {}): AstroIntegration {
 	return {
 		name: '@astrojs/preact',
-		clientEntrypoint: development ? '@astrojs/preact/client-dev.js' : '@astrojs/preact/client.js',
-		serverEntrypoint: '@astrojs/preact/server.js',
-		jsxImportSource: 'react',
-		jsxTransformOptions: async () => {
-			// @ts-expect-error types not found
-			const plugin = await import('@babel/plugin-transform-react-jsx');
-			const jsx = plugin.default?.default ?? plugin.default;
-			return {
-				plugins: [
-					jsx({}, { runtime: 'automatic', importSource: 'preact/compat' }),
-					[
-						'babel-plugin-module-resolver',
-						{
-							alias: {
-								react: 'preact/compat',
-								'react-dom/test-utils': 'preact/test-utils',
-								'react-dom': 'preact/compat',
-								'react/jsx-runtime': 'preact/jsx-runtime',
-							},
-						},
-					],
-				],
-			};
-		},
-	};
-}
+		hooks: {
+			'astro:config:setup': ({ addRenderer, updateConfig, command }) => {
+				const preactPlugin = preact({
+					include,
+					exclude,
+					babel: {
+						cwd: fileURLToPath(babelCwd)
+					}
+				});
 
-function getViteConfiguration(compat?: boolean): ViteUserConfig {
-	const viteConfig: ViteUserConfig = {
-		optimizeDeps: {
-			include: ['@astrojs/preact/client.js', 'preact', 'preact/jsx-runtime'],
-			exclude: ['@astrojs/preact/server.js'],
-		},
-	};
+				const viteConfig: ViteUserConfig = {
+					optimizeDeps: {
+						include: ['@astrojs/preact/client.js', 'preact', 'preact/jsx-runtime'],
+						exclude: ['@astrojs/preact/server.js'],
+					},
+				};
 
-	if (compat) {
-		viteConfig.optimizeDeps!.include!.push(
-			'preact/compat',
-			'preact/test-utils',
-			'preact/compat/jsx-runtime'
-		);
-		viteConfig.resolve = {
-			alias: [
-				{ find: 'react', replacement: 'preact/compat' },
-				{ find: 'react-dom/test-utils', replacement: 'preact/test-utils' },
-				{ find: 'react-dom', replacement: 'preact/compat' },
-				{ find: 'react/jsx-runtime', replacement: 'preact/jsx-runtime' },
-			],
-			dedupe: ['preact/compat', 'preact'],
-		};
-		// noExternal React entrypoints to be bundled, resolved, and aliased by Vite
-		viteConfig.ssr = {
-			noExternal: ['react', 'react-dom', 'react-dom/test-utils', 'react/jsx-runtime'],
-		};
-	}
+				// If not compat, delete the plugin that does it
+				if(!compat) {
+					const pIndex = preactPlugin.findIndex(p => p.name == 'preact:config');
+					if (pIndex >= 0) {
+						preactPlugin.splice(pIndex, 1);
+					}
+				} else {
+					viteConfig.optimizeDeps!.include!.push(
+						'preact/compat',
+						'preact/test-utils',
+						'preact/compat/jsx-runtime',
+					);
+					viteConfig.resolve = {
+						alias: [
+							{ find: 'react', replacement: 'preact/compat' },
+							{ find: 'react-dom/test-utils', replacement: 'preact/test-utils' },
+							{ find: 'react-dom', replacement: 'preact/compat' },
```

Comment:
I think we can remove these as it's [covered by the Preact plugin](https://github.com/preactjs/preset-vite/blob/4dfecb379c17fbb2f442987a8ff95536ff290cbd/src/index.ts#L184-L190)